maud_macros/src/ast.rs

```diff
@@ -217,5 +217,13 @@ pub fn join_ranges<I: IntoIterator<Item = SpanRange>>(ranges: I) -> SpanRange {
 }
 
 pub fn name_to_string(name: TokenStream) -> String {
-    name.into_iter().map(|token| token.to_string()).collect()
+    name.into_iter()
+        .map(|token| {
+            if let TokenTree::Literal(literal) = token {
+                literal.to_string().trim_matches('"').to_string()
```

Comment:
Does `.trim_matches("")` work if the literal is a raw string, or contains escapes?

Reply:
Yes.
Check out [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cecce2b4d6f655546d41fe8650d42add) playground.

Reply:
Is this the wanted behaviour? That the raw string `r#""test""#` should be output as just `test` and not `"test"` as the raw string would imply?

Either way it can easily be changed

Reply:
Strings like `r#""t"es"t""#` would still include the interior `"` like `t"es"t`, se  [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=36a4d54383c8eb412ce7db244b784da8) modified playground

Reply:
Well `trim_matches()" does exactly this: see the [doc](https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim_matches).
If this isn't the wanted behaviour then why use this function in first place.

Still I see your point, I just made the assumption that if `trim_matches()` was used then this behaviour was actually correct.

Reply:
This i the behavior I imagined since attributes starting with `"` does not seem relevant, but I do not intend to force my view of the behavior on the project. If attributes starting or ending in `"` was needed I would change the implementation

Reply:
My bad. Haven't read the previous implementation and thought that `trim_matches` was already used beforehand.

Reply:
My concern was that `r#"test"` would be output as literally `r#"test"`. Because `proc_macro::Literal::to_string()` returns the **source code**, not the final string.

If you add the following test then it should be clear what the problem is.

```rust
#[test]
fn string_literals_in_attribute_names() {
    let result = html! { this r#"@sentence:-is.not"#="false" of-course {} };
    assert_eq!(
        result.into_string(),
        r#"<this @sentence:-is.not="false" of-course></this>"#
    );
}
```

Take a look at the [literal parser](https://github.com/lambda-fairy/maud/blob/d1cfde53b6845c0b58c49b77167a5ce27cc7f11b/maud_macros/src/parse.rs#L198) for ideas on how to handle this correctly.

Reply:
> proc_macro::Literal::to_string() returns the source code, not the final string.
Thank you for the clarification. Fixed in c33ee62d87f7d0d2840c0b124ac424f1c311d353